osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -286,9 +286,10 @@ public void Delete(TModel item)
             using (ContextFactory.GetForWrite())
             {
                 // re-fetch the model on the import context.
-                var foundModel = queryModel().Include(s => s.Files).ThenInclude(f => f.FileInfo).First(s => s.ID == item.ID);
+                var foundModel = queryModel().Include(s => s.Files).ThenInclude(f => f.FileInfo).FirstOrDefault(s => s.ID == item.ID);
 
-                if (foundModel.DeletePending) return;
+                // Test for null since FirstOrDefault will return null if nothing is found
+                if (foundModel == null || foundModel.DeletePending) return;
```

Comment:
While we're here, let's make this Delete function return `false` on early exit (when no operation was performed).

Reply:
ok